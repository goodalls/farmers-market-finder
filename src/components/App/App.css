@import url("https://fonts.googleapis.com/css?family=Open+Sans:400,700,800");
.app {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-template-rows: 200px auto;
  background-color: rgba(0, 0, 0, 0.5); }

.background-wrapper {
  background-image: url("./sven-scheuermeier-58010.jpg");
  background-repeat: no-repeat;
  background-size: cover; }
