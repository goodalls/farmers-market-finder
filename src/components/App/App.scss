@import '../../styles/colors.scss';

.app {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-template-rows: auto auto;
  background-color: rgba(0, 0, 0, .5);
}

.background-wrapper {
  background-image: url('./sven-scheuermeier-58010.jpg');
  background-repeat: no-repeat;
  background-size: cover; 
}


