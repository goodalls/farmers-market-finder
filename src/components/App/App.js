import React, { Component } from 'react';
import { Route, withRouter } from 'react-router-dom';
import * as actions from '../../actions/actions';
import Container from '../Container/Container';
import Favorites from '../Favorites/Favorites';
import Control from '../Control/Control';
import Header from '../Header/Header';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Why from '../WhyQuotes/Why';
import '../../styles/colors.css';
import Card from '../Card/Card';
import Map from '../Map/Map';
import './App.css';

export class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: []
    };
  }

  favorite = (event, market) => {
    //needs to check for dups
    if (market.favorite === false) {
      market.favorite = true;
      this.props.updateFavorites(market);
    } else {
      market.favorite = false;
      this.props.removeFavorite(market);
    }
  };

  render() {
    return (
      <div className="background-wrapper">
        <div className="app">
          <Header />
          <Control />
          <Route exact path="/" component={Why} />
          <Route
            exact
            path="/market-list"
            render={() => {
              return <Container fav={this.favorite} />;
            }}
          />
          <Route exact path="/map-list" component={Map} />
          <Route
            exact
            path="/favorite"
            render={() => {
              return <Favorites fav={this.favorite} />;
            }}
          />
          <Route
            path="/single-market/:id"
            render={({ match }) => {
              const { id } = match.params;
              const singleMarket = this.props.markets.find(
                market => market.id === id
              );

              if (singleMarket) {
                return <Card {...singleMarket} fav={this.favorite} />;
              } else {
                return null;
              }
            }}
          />
        </div>
      </div>
    );
  }
}

App.propTypes = {
  markets: PropTypes.array,
  updateFavorites: PropTypes.func,
  removeFavorite: PropTypes.func
};

export const mapStateToProps = store => ({
  markets: store.markets
});

export const mapDispatchToProps = dispatch => ({
  updateFavorites: (favorite) => dispatch(actions.updateFavorites(favorite)),
  removeFavorite: (favorite) => dispatch(actions.removeFavorite(favorite))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
