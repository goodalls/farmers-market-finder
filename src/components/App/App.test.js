import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {shallow} from 'enzyme';

it.skip('should match the snapshot', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
