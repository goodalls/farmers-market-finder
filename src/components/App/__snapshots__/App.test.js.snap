// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="app"
>
        <Header />
        <withRouter(Connect(Control)) />
        <Route
                component={[Function]}
                exact={true}
                path="/"
        />
        <Route
                exact={true}
                path="/market-list/:id"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/favorite"
                render={[Function]}
        />
        <Route
                path="/single-market/:id"
                render={[Function]}
        />
</div>,
      "className": "background-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <withRouter(Connect(Control)) />,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            exact={true}
            path="/market-list/:id"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/favorite"
            render={[Function]}
/>,
          <Route
            path="/single-market/:id"
            render={[Function]}
/>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/market-list/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/favorite",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/single-market/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="app"
>
          <Header />
          <withRouter(Connect(Control)) />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    exact={true}
                    path="/market-list/:id"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/favorite"
                    render={[Function]}
          />
          <Route
                    path="/single-market/:id"
                    render={[Function]}
          />
</div>,
        "className": "background-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Header />,
            <withRouter(Connect(Control)) />,
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              exact={true}
              path="/market-list/:id"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/favorite"
              render={[Function]}
/>,
            <Route
              path="/single-market/:id"
              render={[Function]}
/>,
          ],
          "className": "app",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/market-list/:id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/favorite",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/single-market/:id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
